pydm/widgets/related_display_button.py

```diff
@@ -602,6 +605,13 @@ def open_display(self, filename, macro_string="", target=None):
             # Not a pydm app: need to give our new display proper pydm styling
             # Usually done in PyDMApplication
             merge_widget_stylesheet(widget=display)
+            # Clean up references to closed subdisplays
+            for old_display in list(self._subdisplays):
+                # isVisible only goes False after clicking "close"
+                if not old_display.isVisible():
```

Comment:
Yes, I think this will happen. I don't think there's a sensible way around this, other than going back to the `setParent` idea and accepting that it's better for screens to close predictably rather than unpredictably.